src/components/time-entries/TimeEntries.tsx

```diff
@@ -1,109 +1,100 @@
 
 import * as Styled from "./TimeEntries.styled";
 
 interface TimeEntriesProps {
-  updateTimeEntries: Function;
+  timeEntries: Types.TimeEntry[];
+  deleteEntry: Function;
+  searchInput: string;
 }
 
-function TimeEntries({ updateTimeEntries }: TimeEntriesProps) {
-  const state = useContext(StoreContext);
-  const [timeEntries] = state.timeEntries;
-
-  async function deleteEntry(id: number) {
-    deleteTimeEntry(id);
-    updateTimeEntries();
-  }
-
-  useEffect(() => {
-    updateTimeEntries();
-  }, []);
-
+function TimeEntries({ timeEntries, deleteEntry, searchInput }: TimeEntriesProps) {
   return (
     <Styled.TimeEntries>
-      {timeEntries.map((timeEntry, index) => {
-        const startTimeObject = new Date(timeEntry.startTime);
-        const startTime = startTimeObject.toLocaleTimeString("nl-NL", {
-          hour: "2-digit",
-          minute: "2-digit",
-        });
-
-        const endTimeObject = new Date(timeEntry.endTime);
-        const endTime = endTimeObject.toLocaleTimeString("nl-NL", {
-          hour: "2-digit",
-          minute: "2-digit",
-        });
-
-        const today = new Date();
-        const yesterday = new Date(today);
-
-        yesterday.setDate(yesterday.getDate() - 1);
-
-        const dateStringToday = today.toLocaleDateString("nl-NL");
-        const dateStringYesterday = yesterday.toLocaleDateString("nl-NL");
-
-        const dateObject = new Date(timeEntry.startTime);
-        const currentDate = dateObject.toLocaleDateString("nl-NL");
-
-        const nextTimeEntry = timeEntries?.[index + 1];
-        const nextDateObject = new Date(nextTimeEntry?.startTime);
-        const nextDate = nextDateObject?.toLocaleDateString("nl-NL");
-
-        const previousTimeEntry = timeEntries?.[index - 1];
-        const previousDateObject = new Date(previousTimeEntry?.startTime);
-        const previousDate = previousDateObject?.toLocaleDateString("nl-NL");
-
-        function isSameDay() {
-          if (currentDate === dateStringToday) {
-            return "(Today)";
+      {timeEntries
+        .filter((timeEntry) => timeEntry.client.toLowerCase().includes(searchInput.toLowerCase()))
+        .map((timeEntry, index) => {
+          const startTimeObject = new Date(timeEntry.startTime);
+          const startTime = startTimeObject.toLocaleTimeString("nl-NL", {
+            hour: "2-digit",
+            minute: "2-digit",
+          });
+
+          const endTimeObject = new Date(timeEntry.endTime);
+          const endTime = endTimeObject.toLocaleTimeString("nl-NL", {
+            hour: "2-digit",
+            minute: "2-digit",
+          });
+
+          const today = new Date();
+          const yesterday = new Date(today);
+
+          yesterday.setDate(yesterday.getDate() - 1);
+
+          const dateStringToday = today.toLocaleDateString("nl-NL");
+          const dateStringYesterday = yesterday.toLocaleDateString("nl-NL");
+
+          const dateObject = new Date(timeEntry.startTime);
```

Comment:
`dateObject` seems a bit too generic of a name between all these other date related variables